### ISSUE TITLE
Overflow behavior isn't documented

### ISSUE THREAD

**Issue by jamesderlin** [dart-lang/fixnum#78]
The documentation for `Int32` and `Int64` state:

> Arithmetic operations may overflow in order to maintain this range.

but doesn't explain what the overflow behavior is.  Does it wrap around?  Is it clamped?  Is it like signed integer "overflow" in C where it's undefined behavior?

In contrast, [the documentation for `int`](https://api.dart.dev/stable/dart-core/int-class.html) states:

> The default implementation of `int` is 64-bit two's complement integers with operations that wrap to that range on overflow.

**Comment by creativecreatorormaybenot** [comment]
@jamesderlin thanks for this issue. I feel like the implementation is likely also not correct in that case because (at least from what I understand), the goal of `Int64` should be to emulate the behavior of Dart's `int` **in native** (just adding web support), but it does not work that way.

Instead, any overflow results in `-2476544402191318784`, unfortunately (so no matter the overflow, you always get that number on web. On native, it works correctly.

**Comment by cbracken** [comment]
> the goal of Int64 should be to emulate the behavior of Dart's int in native (just adding web support), but it does not work that way.

I switched roles at work ~5 years ago and don't work on this package any more, but when we wrote this package, Dart's `int` behaviour on the VM was effectively the same as Python integers. We moved to 64-bit integers long after this package went stable.

> Instead, any overflow of Int64 results in -2476544402191318784, unfortunately (so no matter the overflow, you always get that number on web. On native, it works correctly.

Can you file this as a separate bug and link here? The goal of the package is to have the same behaviour on the web and VM; if that's not happening, then it's a bug. We should add additional cases to the [overflow tests](https://github.com/dart-lang/fixnum/blob/9d90a12c0554554c9f3438d610014c713f8be0d0/test/int64_test.dart#L603-L613)

**Comment by jamesderlin** [comment]
> Instead, any overflow of `Int64` results in `-2476544402191318784`, unfortunately (so no matter the overflow, you always get that number on web. On native, it works correctly.

@creativecreatorormaybenot I don't think that I've observed that.  Do you have a specific reproduction case?
